### PR TITLE
Revised Decision Trees lecture

### PR DESCRIPTION
**Pull request by trgardos**
Moved old lecture to `14-save-Classification-I-Decision-Trees.qmd` and replaced with new, revised lecture.

Expect more tweaks for slide formatting, but this can be merged still.

### CODE REVIEW

File: ds701_book/14-Classification-I-Decision-Trees.ipynb
Comment (sladenheim) [inline ds701_book/14-Classification-I-Decision-Trees.ipynb:1]: I think this file should be in jupyter_books

File: ds701_book/14-Classification-I-Decision-Trees.qmd [hunk 1]
```diff
@@ -1,407 +1,1158 @@
...
+possibly involving combinatorial searching of groupings.  
+
+However for ordinal or continuous attributes, sweeping through a range of
+threshold values can be more efficient.
+
+## Measures for Selecting Attribute and Test Condition
+
+Ideally, we want to pick attributes and test conditions that maximize the homogeneity of the splits.
+
+We can use a impurity index to measure the homogeneity of a split.
```
Comment (sladenheim) [inline ds701_book/14-Classification-I-Decision-Trees.qmd:478]: an impurity

File: ds701_book/14-Classification-I-Decision-Trees.qmd [hunk 2]
```diff
@@ -1,407 +1,1158 @@
...
+
+## Gain Ratio
 
-An __Ordinal__ attribute is similar to a Nominal attribute, except that since
-attributes have an ordering, we can specify the test in terms of a threshold.  
+From [@Tan2018, p. 127]:
 
-This simplifies the search for a good paritition.
+* Having a low impurity value alone is insufficient to find a good attribute test condition for a node. 
+* Having more number of child nodes can make a decision tree more complex and consequently more susceptible to overfitting.
```
Comment (sladenheim) [inline ds701_book/14-Classification-I-Decision-Trees.qmd:710]: I think you were trying to say either:

Having a larger number of child nodes...

OR

Having more child nodes ...

File: ds701_book/14-Classification-I-Decision-Trees.qmd [hunk 3]
```diff
@@ -1,407 +1,1158 @@
...
 
 ---
 
-A __Continuous__ attribute can be handled two ways:
-    
-It can be __discretized__ to form an ordinal categorical attribute
+There are two ways to overcome this problem. 
+
+1. One way is to _generate only binary decision trees_, thus avoiding the difficulty of handling attributes with varying
+   number of partitions. This strategy is employed by decision tree classifiers such as **CART**.
```
Comment (sladenheim) [inline ds701_book/14-Classification-I-Decision-Trees.qmd:719]: Should we give the acronym for CART?

File: ds701_book/14-Classification-I-Decision-Trees.qmd [hunk 4]
```diff
@@ -1,407 +1,1158 @@
...
-Using the "Student ID" attribute yields perfect accuracy on the training data.
+## Ensemble Methods
+
+(See @Tan2018, Chapter 4)
+
+Motivated around the idea that combining several noisy classifers can result in a better prediction
+under certain conditions.
+
+* The base classifiers are independent
+* The base classifiers are noisy (high variance)
```
Comment (sladenheim) [inline ds701_book/14-Classification-I-Decision-Trees.qmd:833]: Should we add a slide about bias and variance?

**Review by sladenheim — APPROVED** [review]
I spotted a few very minor typos.

I added a few thoughts about adding in:
- Explaining the acronym for CART
- a slide mentioning bias and variance, I don't think we have mentioned it before.

**Comment by trgardos** [comment]
Thanks. I'll merge so students have access then we can continue to tweak. Feel free to contribute.

**Pull request merged**